components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-grid-view/nft-grid-view-item.tsx

```diff
@@ -28,21 +32,60 @@ interface Props {
 
 export const NFTGridViewItem = (props: Props) => {
   const { token, onSelectAsset } = props
-  const tokenImageURL = stripERC20TokenImageURL(token.asset.logo)
+  const { asset } = token
+
+  // state
+  const [showMore, setShowMore] = React.useState<boolean>(false)
+  const [showEditModal, setShowEditModal] = React.useState<boolean>(false)
+
+  // methods
+  const onToggleShowMore = React.useCallback((event: React.MouseEvent<HTMLButtonElement>) => {
+    event?.stopPropagation()
+    setShowMore((currentValue) => !currentValue)
+  }, [])
+
+  const onHideModal = React.useCallback(() => {
+    setShowEditModal(false)
+  }, [])
+
+  const onEditNft = React.useCallback(() => {
+    setShowEditModal(true)
+    setShowMore(false)
+  }, [])
+
+  // memos
+  const tokenImageURL = React.useMemo(() => {
+    return stripERC20TokenImageURL(token.asset.logo)
+  }, [token.asset.logo])
```

Comment:
`tokenImageURL` is only used by to compute the `remoteImage` value. We should combine the memos so we only have to memoize one value

Reply:
Combined into one memo